### PR TITLE
Remove plain root password from virt folder

### PR DESCRIPTION
**Pull request by waynechen55**
* **Remove** plain root password from all references in QE Virtualization Functional folders. Root password is stored in ```$testapi::password``` by default.

*  **Remove** unused data/virt_autotest/tw_ipxe/menu.ipxe which also have plain root password in it.

* **In** order to make flexible changes, introducing two new subroutines in ```lib/guest_installation_and_configuration_base.pm:``` ```config_guest_installation_command``` and ```config_guest_plain_password```.

* **Subroutine** config_guest_plain_password should be called in subroutine ```config_guest_installation_command``` to form the final virt-install command which may need plain password to facilitate guest installation.

* **Subroutine** ```config_guest_plain_password``` can nearly help handle all cases that may need different plain password other than plain root password by adding new HOLDERs and new PASSWORDs.

* **Verification runs:**
  * [UEFI 15sp6 on 15sp6 kvm using $testapi::password](https://openqa.suse.de/tests/14505870)
  * [UEFI 15sp6 on 15sp6 xen using $testapi::password](https://openqa.suse.de/tests/14525469)
  * [UEFI 12sp5 on 5sp6 kvm using _SECRET_ROOT_PASSWORD](https://openqa.suse.de/tests/14525344)
  * [UEFI 12sp5 on 15sp6 xen using _SECRET_ROOT_PASSWORD](https://openqa.suse.de/tests/14506393) 
  * [UEFI 15sp5 on 15sp6 xen using $testapi::password](https://openqa.suse.de/tests/14516390)
  * [SEV-ES 15sp5 on 15sp6 kvm _SECRET_ROOT_PASSWORD](https://openqa.suse.de/tests/14525324)
  * [oraclelinux uisng _SECRET_ROOT_PASSWORD](https://openqa.suse.de/tests/14525323)
  * [oraclelinux using $testapi::password](https://openqa.suse.de/tests/14516392)

### CODE REVIEW

**Comment by waynechen55** [comment]
@alice-suse @guoxuguang @Julie-CAO @RoyCai7 @nanzhg @tbaev Please have a look.

**Comment by Julie-CAO** [comment]
LGTM

File: data/virt_autotest/tw_ipxe/menu.ipxe [hunk 1]
```diff
@@ -1,150 +0,0 @@
-# This is the ipxe menu for installing TW on O3
```
Comment (alice-suse) [inline data/virt_autotest/tw_ipxe/menu.ipxe:1]: @Julie-CAO Is it fine to remove this menu.ipxe file?

File: lib/guest_installation_and_configuration_base.pm [hunk 2]
```diff
@@ -2416,6 +2416,71 @@ sub validate_guest_installation_automation_file {
...
+
+=cut
+
+sub config_guest_plain_password {
+    my $self = shift;
+
+    $self->reveal_myself;
+    if ($self->{virt_install_command_line} =~ /ROOTPASSWORD/) {
+        unless ($testapi::password) {
+            my $_root_password = get_required_var('_SECRET_ROOT_PASSWORD');
```
Comment (alice-suse) [inline lib/guest_installation_and_configuration_base.pm:2472]: If this is meant for vm root password, will it be better to make it reflected in the setting name, eg `_SECRET_VM_ROOT_PASSWORD`?

File: data/virt_autotest/tw_ipxe/menu.ipxe (hunk 1 above)
Comment (Julie-CAO) [reply data/virt_autotest/tw_ipxe/menu.ipxe:1]: Yes, Wayne asked me offline. In the past, menu.ipxe on O3 was maintained by me, so I added it in github for tracking purpose, but now it is maintained by tools team, so the file is not required any more.

File: lib/guest_installation_and_configuration_base.pm (hunk 2 above)
Comment (waynechen55) [reply lib/guest_installation_and_configuration_base.pm:2472]: If you want to used different password, you can use for example, ```_SECRET_VM_ROOT_PASSWORD```. But for this case, vm use the same root password as host, so I use ```_SECRET_ROOT_PASSWORD```.  So I will keep using ```_SECRET_ROOT_PASSWORD``` here.

Comment (alice-suse) [reply lib/guest_installation_and_configuration_base.pm:2472]: Ok, fine for current need.

**Comment by alice-suse** [comment]
LGTM

**Pull request merged**